src/features/dashboard/DashboardPage.tsx

```diff
@@ -112,6 +113,31 @@ export default function DashboardPage(): JSX.Element {
       })
     })
     setCards(randomCards)
+
+    const fetchData = async () => {
+      const walletRPC = new WalletRPC()
+      const transactionRPC = new TransactionRPC()
+      const results = await Promise.all([
```

Comment:
Would be great to show a loading state to user and to handle possible error

Reply:
Updated